tests/simple/simple.go

```diff
@@ -94,14 +94,43 @@ func Match(t *spb.SimpleTest, actual *exprpb.ExprValue) error {
 }
 
 // MatchValue returns whether the actual value is equal to the
-// expected value, modulo the following normalization:
-//	1) All floating-point NaN values are equal.
-//	2) Map comparisons ignore order.
+// expected value, modulo ensuring all floating-point NaN values
+// are equal.
 func MatchValue(tag string, expected *exprpb.Value, actual *exprpb.Value) error {
 	// TODO write normalized comparator.
-	// For now, just compare the protos.
-	if !proto.Equal(expected, actual) {
-		return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+	switch expected.GetKind().(type) {
+	case *exprpb.Value_MapValue:
+		// Maps are handled as repeated entries, but the entries need to be
+		// compared using set equality semantics.
+		expectedMap := expected.GetMapValue()
+		actualMap := actual.GetMapValue()
+		if actualMap == nil || expectedMap == nil {
+			return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+		}
+		expectedEntries := expectedMap.GetEntries()
+		actualEntries := actualMap.GetEntries()
+		if len(expectedEntries) != len(actualEntries) {
+			return fmt.Errorf("%s: Eval got [%v], want [%v]", tag, actual, expected)
+		}
+	NEXT_ELEM:
+		for _, expectedElem := range expectedEntries {
+			for _, actualElem := range actualEntries {
+				keyErr := MatchValue(tag, expectedElem.GetKey(), actualElem.GetKey())
+				if keyErr != nil {
+					continue
+				}
+				valErr := MatchValue(tag, expectedElem.GetValue(), actualElem.GetValue())
+				if valErr == nil {
+					continue NEXT_ELEM
+				}
```

Comment:
You can break or return error right here - we've matched the key and the value is wrong.